### PR TITLE
DBTP-747 Make sure log resource policy is applied

### PR DESCRIPTION
**Pull request by WillGibson**
Fixes the bug in https://uktrade.atlassian.net/browse/DBTP-747

I don't think this is the end of the ticket, we need to think about testing, and I'll keep the ticket open for that, but this does fix an actual bug, so I would like to get it merged.

I have tested it by deploying a temporary environment with OpenSearch. I watched it fail, then ran `make-addons` from this code and watch it succeed.

### CODE REVIEW

File: tests/copilot_helper/fixtures/make_addons/expected/environments/overrides/cfn.patches.yml
Comment (WillGibson) [inline tests/copilot_helper/fixtures/make_addons/expected/environments/overrides/cfn.patches.yml:1]: This is no longer how it's done. I deleted the file, no tests failed.

Comment (WillGibson) [inline tests/copilot_helper/fixtures/make_addons/expected/environments/overrides/cfn.patches.yml:1]: This file is not relevant any more.

File: dbt_copilot_helper/templates/env/overrides/stack.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import * as cdk from 'aws-cdk-lib';
+import { aws_cloudformation as cloudformation, aws_logs as logs, Fn } from 'aws-cdk-lib';
 import * as path from 'path';
-import {aws_logs as logs, aws_cloudformation as cloudformation} from 'aws-cdk-lib';
 import resource_policy from './log_resource_policy.json';
-import {readFileSync} from "fs";
-import {parse} from 'yaml';
+import { readFileSync } from "fs";
+import { parse } from 'yaml';
```
Comment (WillGibson) [inline dbt_copilot_helper/templates/env/overrides/stack.ts:6]: I let my IDE tidy up the imports.

File: dbt_copilot_helper/templates/env/overrides/stack.ts [hunk 2]
```diff
@@ -32,7 +32,7 @@ export class TransformedStack extends cdk.Stack {
         const logResourcePolicy = this.template.getResource("LogResourcePolicy") as logs.CfnResourcePolicy;
 
         addonStack.addDependency(logResourcePolicy);
-        logResourcePolicy.policyDocument = JSON.stringify(resource_policy);
+        logResourcePolicy.policyDocument = Fn.sub(JSON.stringify(resource_policy));
```
Comment (WillGibson) [inline dbt_copilot_helper/templates/env/overrides/stack.ts:35]: This is the fix.

**Review by ejayesh — APPROVED** [review]
lgtm

**Pull request merged**